frontstage/views/account/account.py

```diff
@@ -31,8 +38,42 @@ def update_account(session):
     if not form_valid:
         flash('At least one option should be selected.')
         return redirect(url_for('account_bp.get_account'))
-    if form.data['option'] == 'contact_details':
-        return redirect(url_for('account_bp.change_account_details'))
+    else:
+        return redirect(url_for(form_redirect_mapper.get(form.data['option'])))
+
+
+@account_bp.route('/change-password', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def change_password(session):
+    form = ChangePasswordFrom(request.values)
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
+    if request.method == 'POST' and form.validate():
+        username = respondent_details['emailAddress']
+        password = request.form.get('password')
+        new_password = request.form.get('new_password')
+        if new_password == password:
+            return render_template('account/account-change-password.html', form=form,
+                                   errors={"new_password": ["Your new password is the same as your old password"]})
+        bound_logger = logger.bind(email=obfuscate_email(username))
+        bound_logger.info("Attempting to find user in auth service")
+        try:
+            auth_controller.sign_in(username, password)
```

Comment:
Holy moly this line needs a comment!  Something like 
`We call the sign in function as it's the only way to verify that the password provided is correct`

That wasn't clear when I first read it, and a comment would save someone from having to wonder why we're calling a sign_in function at a glance.

Reply:
ahh.. I thought it was standard to validate a password via signin. Ok will add the comment.

Reply:
It could be our naming convention now that I think about it.  If `sign_in` (which it doesn't actually do) was called `verify_credentials` (which is what it's actually doing) then a comment wouldn't be needed.  `sign_in` kinda implies a check + action to make you be signed in. All this is, is a quick thumbs up/down on whether or not the credentials are correct